app/services/email_alert_api/params_formatter.rb

```diff
@@ -0,0 +1,34 @@
+module EmailAlertApi
+  module ParamsFormatter
+    # This is equivalent to code introduced into email alert frontend:
+    # https://github.com/alphagov/email-alert-frontend/blob/main/app/services/subscriber_list_params/generate_single_page_list_params_service.rb
+    # to handle subscriber lists with reverse links.
+    # It will only be needed in here temporarily while we migrate specialist topics
+    def document_collection_subscriber_list_params(content_item)
+      {
+        "url" => content_item.base_path,
+        "title" => content_item.title,
+        "content_id" => content_item.content_id,
+        "description" => content_item.description,
+      }.merge(document_collection_links(content_item))
+    end
+
+    def document_collection_links(content_item)
+      return {} unless content_item.document_type == "document_collection"
+
+      { "links" => { "document_collections" => [content_item.content_id] } }
+    end
+
+    def specialist_topic_subscriber_list_params(topic)
+      topic.subscriber_list_search_attributes
+    end
+
+    def taxonomy_topic_subscriber_list_params(content_item)
+      {
+        "title" => content_item["title"],
```

Comment:
Assuming the content_item in this method has the same data type as the content_item passed to `document_collection_subscriber_list_params` in line 7, why do we access title using square brackets here in line 28 and using a dot operator in line 10?

Reply:
Ah yes this is confusing.

In line 7 we are passing an instance of the ContentItem class. So we can call the title method etc on that object. In line 28, we are passing this through, ie it is just a hash. Hence needing the square brackets.

I think I will amend this. Will push a fresh commit shortly. Thanks for the thorough review!